views/datasource.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from common.session import session
+from common.ui_utils import ui_utils
+from parsing.table import table
+
+class datasource():
+    web_session = None
+
+    def __init__(self, web_session):
+        self.web_session = web_session
+        self.web_driver = web_session.web_driver
+
+    def list_datasource(self):
+        elements_list = table(self.web_session).get_middleware_datasources_table()
+
+
+        print (elements_list[2])
+
+    def validate_datasource(self, text):
+        table = []
+        for tr in self.web_driver.find_elements_by_xpath('.//tr'):
```

Comment:
should be using ui_utils.get_generic_table_as_dict, and then iterating thru that table.